apps/api/src/controllers/post.ts

```diff
@@ -1,125 +1,117 @@
 import { RequestHandler, Request, Response } from 'express';
-import { categories } from './category';
-import { IPost, IPostComment, IPostRequest, IPostResponse } from '../models/interfaces';
-import { getArrayItemByField, getArrayItemIndexByField } from './utils';
-
-export const posts: IPost[] = [];
-const comments: IPostComment[] = [];
-
-const getPost = (req: Request, res: Response) => {
-    res.status(200).json(posts);
+import Post from '../models/post';
+import PostComment from '../models/postComment';
+
+const getPost = async (req: Request, res: Response) => {
+    try {
+        const posts = await Post.find().populate('category');
+        res.status(200).json(posts);
+    } catch (error) {
+        const { message } = error;
+        res.status(500).json({ message });
+    }
 };
 
-const createPost: RequestHandler = (req: Request, res: Response) => {
-    const { title, image, description, category } = req.body;
-    if (!title) {
-        return res.status(400).json({ message: 'The title is required.' });
-    }
-    if (!image) {
-        return res.status(400).json({ message: 'The image is required.' });
-    }
-    if (!description) {
-        return res.status(400).json({ message: 'The description is required.' });
-    }
-    if (!category) {
-        return res.status(400).json({ message: 'The category is required.' });
+const createPost: RequestHandler = async (req: Request, res: Response) => {
+    try {
+        const newPost = await Post.create(req.body);
+        res.status(201).json(newPost);
+    } catch (error) {
+        const { message } = error;
+        res.status(500).json({ message });
     }
-
-    const newPost: IPost = {
-        id: Date.now().toString(),
-        title,
-        image,
-        description,
-        category,
-        comments: []
-    };
-    posts.push(newPost);
-
-    res.status(201).json(newPost);
 };
 
-const getPostByCategory: RequestHandler = (req: Request, res: Response) => {
+const getPostByCategory: RequestHandler = async (req: Request, res: Response) => {
     const { categoryId } = req.params;
 
-    const postsByCategory: IPost[] = posts.filter(item => item.category === categoryId);
-    
-    postsByCategory.map(postItem => postItem.category = getArrayItemByField(categories, 'id', postItem.category));
-
-    res.status(200).json(postsByCategory);
+    try {
+        const posts = await Post.where('category').equals(categoryId).populate('category');
```

Comment:
Another approach to find the post by category can be `Post.find({ category: categoryId }).populate('category');`